src/deploy/functions/backend.ts

```diff
@@ -255,6 +257,9 @@ export type Endpoint = TargetIds &
     // on GCFv2 always
     uri?: string;
     sourceUploadUrl?: string;
+    // GCFv1 can be http or https and
+    // GCFv2 is always https
+    securityLevel?: SecurityLevel;
```

Comment:
Hmm I honestly liked the old solution better. I'm a little concerned that `securityLevel` is only relevant to GCFv1 HTTP triggers yet we are proposing here that it's part of any Endpoint.

Should we at lesast annotate this as a temporary hack? IIUC, the way forward is to have a breaking change where we always set HTTP function to HTTPS-only.

Reply:
Would dumping it in `backend.Endpoint.httpsTrigger` fit a bit better? I'll also make a comment about this being temporary and link to the Issue.

Looking at the old solution, deleting the entire `httpsTrigger` object doesn't really match with the rest of the deployment process. This way we're at least utilizing the built-in flow, leveraging `inferDetailsFromExisting`, and actually creating the exact CloudFunction object that should be deployed.

Reply:
One thing I liked about the old way is that it is egregious (and therefore obvious) that it's an hack! 

Regardless, if this is something we intend to clean up in the future, I don't think this is that harmful.